test/integration/boundingbox_camera_plugin.cc

```diff
@@ -0,0 +1,474 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <ignition/common/Filesystem.hh>
+#include <ignition/sensors/Manager.hh>
+#include <ignition/sensors/BoundingBoxCameraSensor.hh>
+
+#include <ignition/rendering/RenderEngine.hh>
+#include <ignition/rendering/RenderingIface.hh>
+#include <ignition/rendering/Scene.hh>
+#include <ignition/rendering/BoundingBoxCamera.hh>
+#include <ignition/msgs.hh>
+
+#include "test_config.h"  // NOLINT(build/include)
+#include "TransportTestTools.hh"
+
+using namespace ignition;
+using namespace rendering;
+
+class BoundingBoxCameraSensorTest: public testing::Test,
+  public testing::WithParamInterface<const char *>
+{
+  // Create a BoundingBox Camera sensor from a SDF and gets a boxes message
+  public: void BoxesWithBuiltinSDF(const std::string &_renderEngine);
+
+  public: void Boxes3DWithBuiltinSDF(const std::string &renderEngine);
+};
+
+/// \brief mutex for thread safety
+std::mutex g_mutex;
+
+/// \brief bounding boxes from the camera
+std::vector<BoundingBox> g_boxes;
+
+/// \brief counter of received boundingboxes msg
+int g_counter = 0;
+
+/// \brief callback to receive 2d boxes from the camera
+void OnNewBoundingBoxes(const msgs::AnnotatedAxisAligned2DBox_V &boxes)
+{
+  g_mutex.lock();
+  g_boxes.clear();
+
+  int size = boxes.annotated_box_size();
+  for (int i = 0; i < size; i++)
+  {
+    BoundingBox box(BoundingBoxType::VisibleBox2D);
+    auto annotated_box = boxes.annotated_box(i);
+    auto axisAlignedBox = annotated_box.box();
+    auto min_corner = axisAlignedBox.min_corner();
+    auto max_corner = axisAlignedBox.max_corner();
+
+    box.size.X(max_corner.x() - min_corner.x());
+    box.size.Y(max_corner.y() - min_corner.y());
+    box.size.Z(0);
+
+    box.center.X(min_corner.x() + box.size.X() / 2);
+    box.center.Y(min_corner.y() + box.size.Y() / 2);
+    box.center.Z(0);
+
+    box.oreintation.Set(0, 0, 0, 0);
+    box.label = annotated_box.label();
+    g_boxes.push_back(box);
+  }
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief callback to receive 3D boxes from the camera
+void OnNew3DBoundingBoxes(const msgs::AnnotatedOriented3DBox_V &boxes)
+{
+  g_mutex.lock();
+  g_boxes.clear();
+
+  int size = boxes.annotated_box_size();
+  for (int i = 0; i < size; i++)
```

Comment:
```suggestion
  for (int i = 0; i < size; ++i)
```
The style guide says to use the prefix operator in for loops: https://ignitionrobotics.org/docs/all/contributing#style-guides